openapi-module/README.md

```diff
@@ -0,0 +1,210 @@
+│   ├── parsers/                # Parsing logic
+│   │   └── openapi-parser.js   # Logic for parsing OpenAPI documents
+│   └── utils/                  # Utility functions
+│       └── generation-utils.js # Utility functions for generation scripts
+├── temp/                       # Temporary files and generated outputs
+├── test/                       # Tests for the module
+│   ├── generators.test.js      # Tests for generators
+│   └── parser.test.js          # Tests for parsers
+├── .gitignore                  # Ignored files and directories
+├── package-lock.json           # npm lock file
+├── package.json                # npm package file
+└── README.md                   # Project documentation
+```
+
+## Installation
+
+Install dependencies by running:
+
+```sh
+npm install
+```
+
+## Configuration
+
+### OpenAI API Key Setup
+
+If you use GPT for scenario generation, set your OpenAI API key as an environment variable.
+
+#### Linux/MacOS:
+
+Add the API key to your shell configuration file:
+
+```bash
+export OPENAI_API_KEY='your-api-key-here'
+```
+
+Reload your shell configuration:
+
+```bash
+source ~/.bashrc  # or source ~/.zshrc, etc.
+```
+
+### Module Configuration (Optional)
+
+The module is pre-configured to work out of the box with sensible defaults. You can modify `src/GPT/config.js` to customize the behavior:
+
+- **useGPT**: Default is `false`. Set to `true` to use GPT for generating `.flex` files.
+- **openaiApiKey**: Set this in your environment if using GPT.
+- **model, maxTokens, temperature**: Pre-set for general use, but adjustable for specific needs.
+- **promptTemplate**: Already tailored to generate useful Flex scenarios. Advanced users can modify it.
+- **outputDir, outputFileName**: Defaults to saving outputs in the `temp` directory with a `.flex` extension.
+
+```javascript
+module.exports = {
+    useGPT: false,
+    openaiApiKey: process.env.OPENAI_API_KEY,
+    model: "gpt-3.5-turbo",
+    maxTokens: 1500,
+    temperature: 0.7,
+    promptTemplate: function(endpoints) {
+        return `
+        You are given the following API endpoints from an OpenAPI document:
+
+        ${JSON.stringify(endpoints, null, 2)}
+
+        Please generate a Flex scenario JSON file that includes:
+        ...
+        `;
+    },
+    outputDir: '../../temp',
+    outputFileName: 'flex-scenario-gpt.flex',
+};
+```
+
+## Usage
+
+### Generating `.flex` Files and cURL Commands
+
+You can generate `.flex` files and cURL commands using the scripts provided.
+
+#### Static Approach (Faker)
+
+1. Run the script with `--useGPT=false` to use Faker for generating data.
+2. Customize field mappings in `field-mapping.js` if needed.
+3. Run the scripts.
+
+#### AI Approach (OpenAI GPT)
+
+1. Run the script with `--useGPT=true` to use GPT for generating data.
+2. Ensure your API key is set and the `promptTemplate` is configured in `config.js`.
+3. Run the scripts.
+
+### Running the Scripts
```

Comment:
Under this line we should note that all the following commands should be running under folder openapi-module